integrations/opennms-wsman-asset-provisioning-adapter/src/main/java/org/opennms/netmgt/provision/WsManAssetProvisioningAdapter.java

```diff
@@ -0,0 +1,463 @@
+                LOG.info("createScheduleForNode: Scheduling {} for nodeid {} with schedule: {}", adapterOperationType, nodeId, aos);
+                return aos;
+        }
+
+        @Override
+        public boolean isNodeReady(AdapterOperation op) {
+                return true;
+        }
+
+        /**
+         * <p>doAdd</p>
+         *
+         * @param nodeId a int.
+         * @param retry a boolean.
+         * @throws org.opennms.netmgt.provision.ProvisioningAdapterException if any.
+         */
+        @Override
+        public void doAddNode(final int nodeId) throws ProvisioningAdapterException {
+                LOG.debug("doAdd: adding nodeid: {}", nodeId);
+
+                final OnmsNode node = m_nodeDao.get(nodeId);
+                Assert.notNull(node, "doAdd: failed to return node for given nodeId:"+nodeId);
+
+                InetAddress ipaddress = m_template.execute(new TransactionCallback<InetAddress>() {
+                        @Override
+                        public InetAddress doInTransaction(TransactionStatus arg0) {
+                                return getIpForNode(node);
+                        }
+                });
+                String vendor = m_template.execute(new TransactionCallback<String>() {
+                        @Override
+                        public String doInTransaction(TransactionStatus arg0) {
+                                 LOG.debug("doAdd: Fetching vendor asset string");
+                                return node.getAssetRecord().getVendor();
+                        }
+                });
+                LOG.debug("doAdd: Fetched asset string: " + vendor);
+
+                if (m_wsManConfigDao == null) {
+                        m_wsManConfigDao = BeanUtils.getBean("daoContext", "wsManConfigDao", WSManConfigDao.class);
+                }
+                final WsmanAgentConfig config = m_wsManConfigDao.getAgentConfig(ipaddress);
+                final WSManEndpoint endpoint = WSManConfigDao.getEndpoint(config, ipaddress);
+                final WSManClient client = m_factory.getClient(endpoint);
+                LOG.debug("doAdd: m_config: " + m_config);
+
+                final OnmsAssetRecord asset = node.getAssetRecord();
+                m_config.getReadLock().lock();
+                try {
+                    for (final AssetField field : m_config.getAssetFieldsForAddress(ipaddress, vendor)) {
+                        try {
+                                final String value = fetchWsManAssetString(client, endpoint, field.getWqlObjs(), field.getFormatString());
+                                LOG.debug("doAdd: Setting asset field \" {} \" to value: {}", field.getName(), value);
+                                // Use Spring bean-accessor classes to set the field value
+                                final BeanWrapper wrapper = PropertyAccessorFactory.forBeanPropertyAccess(asset);
+                                try {
```

Comment:
One to many try statements here. Catch the same exception, but in the earlier try block.

From line 174 to 179 should be this instead:
```
wrapper.setPropertyValue(field.getName(), value);

                    } catch (final BeansException e) {
                        LOG.warn("doAdd: Could not set property \" {} \" on asset object {}", field.getName(),
                                e.getMessage(), e);
                    } catch (final Throwable t) {
```

Reply:
Like so?  I didn't realize catches were "stackable". (IANAJD, clearly)
```
                    for (final AssetField field : m_config.getAssetFieldsForAddress(ipaddress, vendor)) {
                        try {
                                final String value = fetchWsManAssetString(client, endpoint, field.getWqlObjs(), field.getFormatString());
                                LOG.debug("doAdd: Setting asset field \" {} \" to value: {}", field.getName(), value);
                                // Use Spring bean-accessor classes to set the field value
                                final BeanWrapper wrapper = PropertyAccessorFactory.forBeanPropertyAccess(asset);
                        } catch (final BeansException e) {
                                LOG.warn("doAdd: Could not set property \" {} \" on asset object {}", field.getName(),
                                e.getMessage(), e);
                        } catch (final Throwable t) {
                                // This exception is thrown if the WSMAN ENUM fails or an incorrect number of
                                // parameters is returned by the agent or because of a misconfiguration.
                                LOG.warn("doAdd: Could not set value for asset field \" {} \": {}", field.getName(), t.getMessage(), t);
                        }
                }

```

Reply:
Yep, that looks better!